### PR TITLE
Feature: Changes for Recipe Steps

### PR DESCRIPTION
**Pull request by andre-meyer**

### CODE REVIEW

File: src/components/Link/Link.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export interface LinkProps extends HTMLAttributes<HTMLAnchorElement> {
 
 export const Link = ({ children, href, target = '_blank', iconLeft, iconRight, ...other }: LinkProps) => {
   return (
-    <StitchedLink target={target} {...other}>
+    <StitchedLink target={target} href={href} {...other}>
```
Comment (oskarvu) [inline src/components/Link/Link.tsx:16]: Isn't href already in `{...other}`?

File: src/components/Steps/steps/HeaderStep.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  justifyContent: 'flex-start',
+  alignItems: 'center',
+  gap: 4,
+  marginBottom: 8,
+  width: '100%',
+})
+
+export default HeaderStep
+
+/*
```
Comment (oskarvu) [inline src/components/Steps/steps/HeaderStep.tsx:50]: Do we need this comment?

File: src/providers/MetadataFetchProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import React, { createContext, ReactNode, useCallback, useContext, useState } from 'react'
```
Comment (oskarvu) [inline src/providers/MetadataFetchProvider.tsx:1]: Couldn't we just use hook at V2 to get metadata for tokens, and make components at ui-library dumb, so they only take care of displaying stuff? I feel more comfortable with just creating hook for fetching metadata at V2 (one is already created and used at input dialog). At v2 we can also use existing status types and payload management. What do you think?

File: src/components/Steps/steps/HeaderStep.tsx (hunk 2 above)
Comment (andre-meyer) [reply src/components/Steps/steps/HeaderStep.tsx:50]: Forgot to remove, sorry :D

File: src/providers/MetadataFetchProvider.tsx (hunk 3 above)
Comment (andre-meyer) [reply src/providers/MetadataFetchProvider.tsx:1]: Discussed in Discord, will go ahead with this for now and possibly change later after discussing in a bigger group

**Review by oskarvu — APPROVED** [review]

**Pull request merged**